types/sort-array/index.d.ts

```diff
@@ -0,0 +1,16 @@
+// Type definitions for sort-array 2.0
+// Project: https://github.com/75lb/sort-array#readme
+// Definitions by: mrmlnc <https://github.com/mrmlnc>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+type ColumnNames<T> = keyof T | Array<keyof T>;
+type CustomOrderTypes<T> = Array<T[keyof T]>;
+type CustomOrder<T> = Record<keyof T, CustomOrderTypes<T>>;
+
+declare function sortArray<T>(recordset: T[], columnNames: ColumnNames<T>, customOrder?: Partial<CustomOrder<T>>): T[];
+declare function sortArray(recordset: object[], columnNames: string | string[], customOrder?: Record<string, any[]>): object[];
+
+declare namespace sortArray { }
```

Comment:
This hack shouldn't be here, and importers should use `import sortArray = require('sort-array')` instead.

Reply:
Yeap! I didn't know about it. Now I read it here https://github.com/DefinitelyTyped/DefinitelyTyped/pull/21433#discussion_r150398140. Will be fixed now.

Reply:
@DanielRosenwasser, fixed in all my PP's.